chore: add requires docs for cosmo connect (#259)
wundergraph/cosmo-docs: docs/federation/directives/external.mdx

---
title: "@external"
icon: "arrow-right-from-bracket"
description: "The @external directive declares that a field is not independently resolvable by the current subgraph, making it available for use with @requires and @provides."
---

## Definition

```graphql
directive @external on FIELD_DEFINITION | OBJECT
```

## Overview

The `@external` directive declares that a field is not independently resolvable by the current subgraph. The exact behavior depends on where the field is referenced:

> - **Unresolvable by this subgraph** — An `@external` field referenced only in a [`@requires`](/federation/directives/requires) `FieldSet`, or defined only to satisfy an interface. The router will never ask this subgraph to resolve it. The field exists locally so `@requires` can depend on its value, or so the type satisfies its interface contract.

Reviewer: ```suggestion
- **Unresolvable by this subgraph** — An `@external` field referenced only in a [`@requires`](/federation/directives/requires) `FieldSet`, or defined only to satisfy an interface. The router will never consider this subgraph to resolve it. The field is defined locally so that `@requires` can reference it in its `FieldSet` or to satisfy an interface contract.
```
